### ISSUE TITLE
[Object Detection] Upgrade to TFJob v1alpha2

### ISSUE THREAD

**Issue by jlewi** [kubeflow/examples#179]
We should upgrade the object detection dexample to use Kubeflow 0.2 and TFJob v1alpha2.

https://github.com/kubeflow/examples/blob/master/object_detection/jobs/pets-training.yaml

**Comment by royxue** [comment]
@jlewi 
I could make a PR of this as well, as I was trying to make this example to work today. 

This yaml cannot work well for me right now.
It could successfully created the tfjob, I could use `get tfjobs` to see it, but there is no any related pod started, strange.

**Comment by royxue** [comment]
Here is my modified yaml, changed apiVersion, add GPU resources, re-format the volumes/volumeMounts values.
```
---
apiVersion: kubeflow.org/v1alpha2
kind: TFJob
metadata:
  name: pets-training
spec:
  replicaSpecs:
  - replicas: 1
    template:
      spec:
        containers:
        - image: lcastell/pets_object_detection
          resources:
            limits:
              nvidia.com/gpu: 1
          imagePullPolicy: IfNotPresent
          name: tensorflow
          workingDir: /models
          command:
            - "python"
            - "research/object_detection/train.py"
          args:
            - "--logtostderr"
            - "--pipeline_config_path=/pets_data/faster_rcnn_resnet101_pets.config"
            - "--train_dir=/pets_data/train"
          volumeMounts:
            - mountPath: "/pets_data"
              name: pets-data
        volumes:
          - name: pets-data
            persistentVolumeClaim:
              claimName: pets-data-claim
        restartPolicy: OnFailure
    tfReplicaType: MASTER
  - replicas: 1
    template:
      spec:
        containers:
        - image: lcastell/pets_object_detection
          resources:
            limits:
              nvidia.com/gpu: 1
          imagePullPolicy: IfNotPresent
          name: tensorflow
          command:
            - "python"
            - "research/object_detection/train.py"
          args:
            - "--logtostderr"
            - "--pipeline_config_path=/pets_data/faster_rcnn_resnet101_pets.config"
            - "--train_dir=/pets_data/train"
          volumeMounts:
            - mountPath: "/pets_data"
              name: pets-data
        volumes:
          - name: pets-data
            persistentVolumeClaim:
              claimName: pets-data-claim
        restartPolicy: OnFailure
    tfReplicaType: WORKER
  - replicas: 1
    template:
      spec:
        containers:
        - image: lcastell/pets_object_detection
          resources:
            limits:
              nvidia.com/gpu: 1
          imagePullPolicy: IfNotPresent
          name: tensorflow
          command:
            - "python"
            - "research/object_detection/train.py"
          args:
            - "--logtostderr"
            - "--pipeline_config_path=/pets_data/faster_rcnn_resnet101_pets.config"
            - "--train_dir=/pets_data/train"
          volumeMounts:
            - mountPath: "/pets_data"
              name: pets-data
        volumes:
          - name: pets-data
            persistentVolumeClaim:
              claimName: pets-data-claim
        restartPolicy: OnFailure
    tfReplicaType: PS

```

@jlewi  could you help taking a look?

**Comment by ldcastell** [comment]
Hi @royxue, is this last yaml you posted working for you?

@jlewi, do you think it will be better to address this change in issue #178? I can make the change to the yaml quickly but just wanted to confirm with you :)

Thanks!

**Comment by royxue** [comment]
@ldcastell 
Nope, there are some api changes in the v1alpha2, I will make some changes and show you later

**Comment by jlewi** [comment]
It looks like this has been fixed as part of #178 so closing this issue.
